policy/nomad/parser.go

```diff
@@ -103,19 +102,9 @@ func canonicalizePolicy(from *api.ScalingPolicy, to *policy.Policy) {
 
 	if to.Source == "" {
 		to.Source = plugins.InternalAPMNomad
+	}
 
-		// TODO(luiz) move default query logic handling to the Nomad APM plugin
-		parts := strings.Split(to.Query, "_")
-		op := parts[0]
-		metric := parts[1]
-
-		switch metric {
-		case "cpu":
-			metric = "nomad.client.allocs.cpu.total_percent"
-		case "memory":
-			metric = "nomad.client.allocs.memory.usage"
-		}
-
-		to.Query = fmt.Sprintf("%s/%s/%s/%s", metric, from.Target["Job"], from.Target["Group"], op)
+	if to.Source == plugins.InternalAPMNomad {
+		to.Query = fmt.Sprintf("%s/%s/%s", to.Query, from.Target["Group"], from.Target["Job"])
```

Comment:
curious if this needs to be its own conditional or could it be moved above?

Reply:
This needs to be a separate conditional for the case when `to.Source` is defined by the operator as already being `plugins.InternalAPMNomad`, for example:

```hcl
...
scaling {
  ...
  policy {
    source = "nomad-apm"
    query  = "avg_cpu"
    ...
  }
}
```

It's a bit of an edge case, but we should cover it regardless. I am writing tests for the parser in a separate PR, so I will make sure to create one to cover this.

Reply:
ohhh yeah!